ui/settings.lua

```diff
@@ -3,70 +3,127 @@ local S = minetest.get_translator("mail")
 
 local FORMNAME = "mail:settings"
 
-local date_formats = {"%Y-%m-%d %X", "%d/%m/%y %X", "%A %d %B %Y %X"}
+local function deepcopy(orig)
+    local orig_type = type(orig)
+    local copy
+    if orig_type == 'table' then
+        copy = {}
+        for orig_key, orig_value in next, orig, nil do
+            copy[deepcopy(orig_key)] = deepcopy(orig_value)
+        end
+        setmetatable(copy, deepcopy(getmetatable(orig)))
+    else -- number, string, boolean, etc
+        copy = orig
+    end
+    return copy
+end
```

Comment:
Is there a reason to basically implement `table.copy` with questionable key extension here or could `table.copy` be used instead?
I really hope using tables as keys isn't reason to do this here...

Reply:
If I don't do a `deepcopy`, the duplicated table from data isn't really a duplicate but is a pointer, and modify the entire data while I want it just takes that data for displaying. Yep, lua tables sucks...

Reply:
`table.copy` does deep copy of table, it isn't pointer to original table.

Reply:
I can't find any `table.copy` function.

Reply:
https://raw.githubusercontent.com/minetest/minetest/5.7.0/doc/lua_api.txt

Reply:
Done with 6b8ce29